crates/e2e/macro/src/codegen.rs

```diff
@@ -130,23 +135,10 @@ impl InkE2ETest {
             quote! { #wasm_path }
         });
 
-        const DEFAULT_CONTRACTS_NODE: &str = "substrate-contracts-node";
-
-        // use the user supplied `CONTRACTS_NODE` or default to `substrate-contracts-node`
-        let contracts_node: &'static str =
-            option_env!("CONTRACTS_NODE").unwrap_or(DEFAULT_CONTRACTS_NODE);
-
-        // check the specified contracts node.
-        if which::which(contracts_node).is_err() {
-            if contracts_node == DEFAULT_CONTRACTS_NODE {
-                panic!(
-                    "The '{DEFAULT_CONTRACTS_NODE}' executable was not found. Install '{DEFAULT_CONTRACTS_NODE}' on the PATH, \
-                    or specify the `CONTRACTS_NODE` environment variable.",
-                )
-            } else {
-                panic!("The contracts node executable '{contracts_node}' was not found.")
-            }
-        }
+        let client_building = match self.test.config.backend() {
+            Backend::Full => build_full_client(&environment, contracts),
+            Backend::RuntimeOnly => todo!("Runtime-only backend not yet supported"),
```

Comment:
Since the runtime only backend is not yet available. I think it is reasonable to postpone the PR and introduce the changes once the runtime only backend is available.

Reply:
Fair point, however this way we will be able to validate easily the next PR - the next code I'll bake is `impl E2EBackend for DrinkClient`. Having already the possibility to run e2e tests with new backend will make testing these changes straightforward.

Reply:
140 lines is not a big change. I think we can squeeze it in the next PR, when you implement the backend for drink. That way we'll have a fuller picture.